addons/project_mrp_account/models/stock_move.py

```diff
@@ -12,6 +12,12 @@ def _get_analytic_distribution(self):
         distribution = self.raw_material_production_id.project_id._get_analytic_distribution()
         return distribution or super()._get_analytic_distribution()
 
+    def _prepare_analytic_line_values(self, account_field_values, amount, unit_amount):
+        res = super()._prepare_analytic_line_values(account_field_values, amount, unit_amount)
+        if res and self.raw_material_production_id:
```

Comment:
I think you can just let the second condition :)
```suggestion
        if self.raw_material_production_id:
```